src/s3/util/fileViewerManager.ts

```diff
@@ -78,86 +92,135 @@ export class S3FileViewerManager {
                     return
                 }
                 await this.removeAndCloseTab(activeTab)
-                await this.openTab(fileNode)
+                await this.openInReadMode(fileNode)
                 this._onDidChange.fire(activeTab.s3Uri)
             }
         })
+    }
 
-        this.window.onDidChangeVisibleTextEditors(editors => {
-            const editorSet = new Set(editors)
-            for (const value of this.activeTabs.values()) {
-                //if visible text editors don't contain a given S3Tab anymore,
-                //set the S3Tab.editor to undefined
-                if (value.editor) {
-                    if (!editorSet.has(value.editor)) {
-                        value.editor = undefined
-                    }
-                }
-            }
-        })
+    public async focusAndCloseTab(
+        uri: vscode.Uri,
+        editor?: vscode.TextEditor,
+        workspace = vscode.workspace
+    ): Promise<void> {
+        if (!editor) {
+            const doc = await workspace.openTextDocument(uri)
+            await this.window.showTextDocument(doc, {
+                preview: false,
+            })
+        } else {
+            await this.window.showTextDocument(editor.document, {
+                preview: false,
+                viewColumn: editor.viewColumn,
+            })
+        }
 
-        vscode.workspace.onDidCloseTextDocument(async closedTextDoc => {
-            if (this.activeTabs.has(closedTextDoc.uri.fsPath)) {
-                const closedTab = this.activeTabs.get(closedTextDoc.uri.fsPath)
-                closedTab!.editor = undefined
-            }
-        })
+        await vscode.commands.executeCommand('workbench.action.closeActiveEditor')
     }
 
     /**
      * Given an S3FileNode, this function:
      * Checks and creates a cache to store downloads
      * Retrieves previously cached files on cache and
      * Downloads file from S3 ands stores in cache
-     * Opens the tab on read-only with the use of an S3Tab
+     * Opens the tab on read-only with the use of an S3Tab, or shifts focus to an edit tab if any.
      *
      * @param fileNode
-     * @returns
      */
-    public async openTab(fileNode: S3FileNode): Promise<void> {
+    public async openInReadMode(fileNode: S3FileNode): Promise<void> {
+        if (fileNode.file.sizeBytes! < SIZE_LIMIT) {
```

Comment:
Nit: you're checking `fileNode.file.sizeBytes! < SIZE_LIMIT` twice in this function. Can this block be merged with the other one?

Reply:
That is for to update the next file to display when a race condition happens, we want to check before hitting `getFile()`  and we also want to recheck before displaying the file